python/deltalake/table.py

```diff
@@ -428,10 +428,19 @@ def history(self, limit: Optional[int] = None) -> List[Dict[str, Any]]:
         :param limit: the commit info limit to return
         :return: list of the commit infos registered in the transaction log
         """
-        return [
-            json.loads(commit_info_raw)
-            for commit_info_raw in self._table.history(limit)
-        ]
+        if (limit is None) or (self.version() + 1 <= limit):
+            start = 0
```

Comment:
It seems like this could be incorrect if `limit` is None and there are log files in the beginning that were deleted, right?

Although IIRC, our history function can't look past the checkpoints at the moment?

Reply:
> It seems like this could be incorrect if `limit` is None and there are log files in the beginning that were deleted, right?
> 
> Although IIRC, our history function can't look past the checkpoints at the moment?

Hm I wasn't aware of this possibility. Then this won't work for that case. I'll check how I can make it work otherwise, maybe it should be reverse order and counting down

Reply:
@wjones127 @rtyler I added a reverse countdown, this actually also fixes immediately: https://github.com/delta-io/delta-rs/issues/1680

However! It relies on using `DeltaTable.version()` which means in these two cases the count won't be correct:

- A user does `DeltaTable.load_version( old_ver )`, then this version is used to count downwards
- Someone writes to the deltalake, without using `write_deltalake `and passing the `DeltaTable `object. 

In any of these two cases the highest version number of the table can't be fetched from the API. I think this can't be fixed until we have a function that fetches the highest available table version.

Reply:
@rtyler @wjones127 the counting will be correct now for all cases and in reversed order. 🕺